test/open_api/map_test.exs

```diff
@@ -0,0 +1,403 @@
+defmodule OpenApiSpex.OpenApi.MapTest do
+  use ExUnit.Case
+  use Plug.Test
+
+  alias OpenApiSpex.OpenApi
+
+  describe "OpenApiSpex.OpenApi.Map.decode/1" do
+    test "OpenApi" do
+      # NOTE: This test could be split into many smaller tests, that is the goal!
+      # TODO: Move the spec to a setup below, where we could easily try a yaml version also
+      spec =
+        "./test/support/encoded_schema.json"
+        |> File.read!()
+        |> Jason.decode!(keys: :atoms)
+        |> OpenApiSpex.OpenApi.Map.decode()
+
+      # OpenAPI Object
+      assert %OpenApi{
+               openapi: openapi,
+               info: info,
+               servers: servers,
+               paths: paths,
+               components: components,
+               security: security,
+               tags: tags,
+               externalDocs: externalDocs,
+               extensions: extensions
+             } = spec
+
+      assert "3.0.0" == openapi
+
+      # OpenApi Info Object
+      assert %OpenApiSpex.Info{
+               title: _title,
+               version: _version,
+               description: _description,
+               contact: contact,
+               license: license
+             } = info
+
+      # OpenApi Contact Object
+      assert %OpenApiSpex.Contact{} = contact
+
+      # OpenApi Contact Object
+      assert %OpenApiSpex.License{} = license
+
+      # OpenApi Specification Extensions
+      assert %{
```

Comment:
`:extensions` should be a map with string keys: https://github.com/GhostGroup/open_api_spex/blob/63858f6dbad5b8c1bab14e70ba616ddd0ff2cb4f/lib/open_api_spex/open_api.ex#L46